.github/workflows/lint.yml

```
@@ -1,45 +1,48 @@
+            ${{ runner.os }}-cargo-
+
+      - name: Install dependencies and build
         run: |
-          python -m venv ~/.venv/ci-venv
-          source ~/.venv/ci-venv/bin/activate
           python -m pip install --upgrade pip
           pip install -r python/requirements-dev.txt
           make develop
+
       - name: Linting with ruff
         run: |
```

actionlint

shellcheck reported issue in this script: SC2046:warning:1:19: Quote this to prevent word splitting

(shellcheck)